client/fingerprint/fingerprint.go

```diff
@@ -92,8 +91,8 @@ type Factory func(*log.Logger) Fingerprint
 // many of them can be applied on a particular host.
 type Fingerprint interface {
 	// Fingerprint is used to update properties of the Node,
-	// and returns if the fingerprint was applicable and a potential error.
-	Fingerprint(*config.Config, *structs.Node) (bool, error)
+	// and returns a diff of updated node attributes and a potential error.
+	Fingerprint(*cstructs.FingerprintRequest, *cstructs.FingerprintResponse) error
```

Comment:
Why not make this be `Fingerprint(*structs.FingerPrintRequest) (structs.FingerprintResponse, error)` so it returns the response?

Reply:
Sure, I think it could be either way. I'm curious what @dadgar thinks as we talked about this signature before.

Reply:
I suggested this so it will be easier to move to a remote RPC

Reply:
For when this moves to a plugin? Makes sense

Reply:
Yep